backend/src/main/java/com/bakdata/conquery/resources/admin/rest/AdminDatasetProcessor.java

```diff
@@ -0,0 +1,301 @@
+package com.bakdata.conquery.resources.admin.rest;
+
+import com.bakdata.conquery.apiv1.FilterSearch;
+import com.bakdata.conquery.io.jackson.InternalOnly;
+import com.bakdata.conquery.io.jackson.Jackson;
+import com.bakdata.conquery.io.storage.MetaStorage;
+import com.bakdata.conquery.io.storage.NamespaceStorage;
+import com.bakdata.conquery.models.concepts.Concept;
+import com.bakdata.conquery.models.concepts.Connector;
+import com.bakdata.conquery.models.concepts.StructureNode;
+import com.bakdata.conquery.models.concepts.select.concept.UniversalSelect;
+import com.bakdata.conquery.models.concepts.select.concept.specific.EventDurationSumSelect;
+import com.bakdata.conquery.models.concepts.tree.ConceptTreeConnector;
+import com.bakdata.conquery.models.concepts.tree.TreeConcept;
+import com.bakdata.conquery.models.config.ConqueryConfig;
+import com.bakdata.conquery.models.datasets.*;
+import com.bakdata.conquery.models.exceptions.JSONException;
+import com.bakdata.conquery.models.exceptions.ValidatorHelper;
+import com.bakdata.conquery.models.identifiable.Identifiable;
+import com.bakdata.conquery.models.identifiable.ids.specific.*;
+import com.bakdata.conquery.models.identifiable.mapping.PersistentIdMap;
+import com.bakdata.conquery.models.jobs.ImportJob;
+import com.bakdata.conquery.models.jobs.JobManager;
+import com.bakdata.conquery.models.jobs.SimpleJob;
+import com.bakdata.conquery.models.messages.namespaces.specific.*;
+import com.bakdata.conquery.models.messages.network.specific.AddWorker;
+import com.bakdata.conquery.models.messages.network.specific.RemoveWorker;
+import com.bakdata.conquery.models.worker.DatasetRegistry;
+import com.bakdata.conquery.models.worker.Namespace;
+import com.bakdata.conquery.models.worker.ShardNodeInformation;
+import com.google.common.base.Strings;
+import com.univocity.parsers.csv.CsvParser;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import javax.validation.Validator;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+
+@Slf4j
+@RequiredArgsConstructor
+@Getter
+public class AdminDatasetProcessor {
+
+	private final MetaStorage storage; // TODO Remove
+	private final ConqueryConfig config;
+	private final Validator validator;
+	@Nullable
+	private final String storagePrefix;
+	private final DatasetRegistry datasetRegistry;
+	private final JobManager jobManager;
+
+
+	public synchronized Dataset addDataset(String name) throws JSONException {
+		// create dataset
+		Dataset dataset = new Dataset();
+		dataset.setName(name);
+
+
+		final List<String> pathName = Strings.isNullOrEmpty(storagePrefix)
```

Comment:
Ist nicht für den PR, aber ich hab mich gefragt, ob wir nicht einfach statt des prefixes zu hudeln und der standalone command in die anderen sachen leaked, es umdrehehen und die storage directory pro command ändern?

Reply:
Ist eine gute Idee